File: Lab 6/C++/Text.cpp
#include "Text.h"

Text::Text(const std::string& str) : line(str) {} // Ініціалізація рядка

std::string Text::getLine() const 
{
    return line;
}

size_t Text::getLength() const
{
    return line.length();
}

size_t Text::countNumbers() const
{
    size_t count = 0;
    for (char ch : line)
    {
        if (ch >= '0' && ch <= '9')  
        {
            ++count;
        }
    }
    return count;
}

File: Lab 6/C#/Text.cs
using System;

public class Text
{
    private string line;

    public Text(string str)
    {
        line = str; // Ініціалізація
    }

    public string GetLine()
    {
        return line; // Повертає рядок
    }

    public int GetLength()
    {
        return line.Length; // Повертає довжину
    }

    public int CountNumbers()
    {
        int count = 0;
        foreach (char ch in line)
        {
            if (char.IsDigit(ch))
            {
                count++; // Лічильник цифр
            }
        }
        return count;
    }
}
